tree/ntuple/v7/inc/ROOT/RNTupleView.hxx

```diff
@@ -139,12 +139,11 @@ protected:
       std::unique_ptr<RFieldBase> field;
       {
          const auto &desc = pageSource->GetSharedDescriptorGuard().GetRef();
-         field = desc.GetFieldDescriptor(fieldId).CreateField(desc);
-      }
-      if constexpr (!std::is_void_v<T>) {
-         if (field->GetTypeName() != RField<T>::TypeName()) {
```

Comment:
> if (field->GetTypeName() != RField<T>::TypeName()) {

This statement would be accurate if the 2 sides were fully I/O normalized (which stripped the typedef except for Double32/Float16).

In addition there is another 'legal' type of casting when two type with distinct spelling are made equal for I/O purpose by a I/O customization rule.

Reply:
No, it would not be accurate: I want to be able to read a `Double32` field back as `float`. And a `std::uint32_t` as `std::uint64_t` etc.